File: backend/Application/Dtos/HRMS/GeofenceResponse.cs
using System;

namespace Application.Dtos.HRMS;

public class GeofenceResponse
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public decimal Latitude { get; set; }
    public decimal Longitude { get; set; }
    public double RadiusInMeters { get; set; }
    public bool IsActive { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
}

File: apps/app/src/data/dtos/attendance/geofence.response.ts
/**
 * @file geofence.response.ts
 * @description Response schema for Geofences.
 * Mirrors the backend GeofenceResponse DTO.
 */

export interface GeofenceResponse {
  id: number;
  name: string;
  latitude: number;
  longitude: number;
  radiusInMeters: number;
  isActive: boolean;
  createdAt: string;
  updatedAt: string | null;
}

export interface GeofencePaginationResponse {
  items: GeofenceResponse[];
  pageNumber: number;
  pageSize: number;
  totalRecords: number;
  totalPages: number;
}

export interface ApiResponseWrapper<T> {
  isSuccess: boolean;
  message: string;
  data: T;
  errors: string[] | null;
}
